### PR TITLE
[Draft] Update 5.1 manual and api folders

### PR DESCRIPTION
**Pull request by shakthimaan**
# Issue Description

The 5.1 manual and API folders were outdated and the latest (December 7, 2023 bb93d...) code has been used to re-generate the documentation.

### CODE REVIEW

File: site/releases/5.1/manual/afl-fuzz.html [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <!DOCTYPE html><html lang="en"><head>
 
 <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
-<meta name="generator" content="hevea 2.36">
+<meta name="generator" content="hevea 2.35">
```
Comment (Octachron) [inline site/releases/5.1/manual/afl-fuzz.html:4]: The difference in the manual part comes from this change of version. Would you mind regenerating the manual with hevea `2.36`?

File: site/releases/index.fr.md [hunk 2]
```diff
@@ -10,6 +10,7 @@ instructions pour installer OCaml par d'autres moyens que la
 compilation des sources, comme par exemple le gestionnaire de paquets
 OPAM et les gestionnaire de paquets spécifiques à une plateforme.
 
+* OCaml [5.1.0](5.1.0.html), [téléchargé](https://github.com/ocaml/ocaml/archive/5.1.0.tar.gz), publiée le 14 septembre, 2023.
```
Comment (Octachron) [inline site/releases/index.fr.md:13]: Good catch, but would you mind adding `5.1` with today as a release date?

**Comment by Octachron** [comment]
Thanks a lot ! 
You are missing an update to the `Changes` file in `site/releases/5.1/notes`.
Except for my three small remarks, the update looks good to me.

**Comment by shakthimaan** [comment]
Thank you for your valuable feedback! I am closing this PR in favour of https://github.com/ocaml/ocaml.org/pull/1854 to server the manuals directly from ocaml.org.

**Comment by Octachron** [comment]
Since I am not sure how long switching to `ocaml/ocaml.org` will take, I will open a PR for patching the manual for 5.1.1 on `v2.ocaml.org` while waiting for the update to avoid propagating incorrect information.

**Pull request closed without merging**